src/ring_theory/coprime/ideal.lean

```diff
@@ -62,9 +62,11 @@ begin
         apply this _ (finset.mem_cons_self _ _), rintro rfl, exact hat hx },
       { have := submodule.coe_mem (μ a), simp only [mem_infi] at this,
         exact this _ (finset.subset_cons _ hb) ab.symm } } },
-  { rintro ⟨hs, Hb⟩, obtain ⟨μ, hμ⟩ := ih.mpr hs,
-    obtain ⟨u, hu, v, hv, huv⟩ := submodule.mem_sup.mp
-      ((eq_top_iff_one _).mp $ sup_infi_eq_top $ λ b hb, Hb b hb $ by { rintro rfl, exact hat hb }),
+  { rintro ⟨hs, Hb⟩,
+    obtain ⟨μ, hμ⟩ := ih.mpr hs,
+    have := sup_infi_eq_top (λ b hb, Hb b hb (ne_of_mem_of_not_mem hb hat).symm),
+    rw [eq_top_iff_one, submodule.mem_sup] at this,
+    obtain ⟨u, hu, v, hv, huv⟩ := this,
```

Comment:
Nice that you managed to dodge the timeout! Do you have any explanation what changed?

Reply:
The old code was quite odd, and seemed to build up a complex term without ever stopping and asking lean to elaborate it. I think the old code happened to work because Lean could work out what to do by unification, but the new code made the problem much harder.

Reply:
Oh right, great!